server/src/main.rs

```diff
@@ -222,7 +229,31 @@ fn main() {
         }
     });
 
-    router.post(r"/api/v1/debug/([0-9]*)/executions/([0-9]*)/stop", debug_execution_stop);
+    let child = child_thread.clone();
+    let kill = kill_flag.clone();
+    router.post(r"/api/v1/debug/([0-9]*)/executions/([0-9]*)/stop", move |req, res, caps| {
+        match debug_execution_stop(caps, child.clone()) {
+            Ok(body) => {
+                let mut child_thread = child.lock().unwrap();
+                kill.1.store(-1, Ordering::Relaxed);
+                if let Some(child) = child_thread.take() {
+                    child.thread.join().unwrap();
+                }
+                send(req, res, &body)
+            },
+            Err(e) => send_error(req, res, e),
+        }.unwrap();
+    });
+
+    let kill = kill_flag.clone();
+    router.post(r"/api/v1/debug/([0-9]*)/executions/([0-9]*)/trace/stop", move |req, res, caps| {
+        match debug_trace_stop(caps, kill.clone()) {
+            Ok(body) => send(req, res, &body),
+            Err(e) => send_error(req, res, e),
+        }.unwrap();
+    });
```

Comment:
Shouldn't need two endpoints here, and all the logic should go inside `debug_execution_stop` rather than the endpoint callback.